### PR TITLE
feat(dashboard): Widget Viewer table pagination

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds pagination buttons to the Widget Viewer table
image

### CODE REVIEW

File: static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx [hunk 1]
```diff
@@ -73,17 +73,23 @@ function transformResult(query: WidgetQuery, result: RawResult): Series[] {
...
   widget: Widget;
+  cursor?: string;
   limit?: number;
+  pagination?: boolean;
 };
 
 type State = {
   errorMessage: undefined | string;
   loading: boolean;
+  pageLinks: undefined | null | string;
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx:92]:
```suggestion
  pageLinks?: null | string;
```

File: static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx [hunk 2]
```diff
@@ -73,17 +73,23 @@ function transformResult(query: WidgetQuery, result: RawResult): Series[] {
...
   organization: OrganizationSummary;
   selection: PageFilters;
   widget: Widget;
+  cursor?: string;
   limit?: number;
+  pagination?: boolean;
 };
 
 type State = {
   errorMessage: undefined | string;
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx:90]:
```suggestion
  errorMessage?: string;
```

File: static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx [hunk 3]
```diff
@@ -142,7 +149,8 @@ class WidgetQueries extends React.Component<Props, State> {
       !isEqual(widget.displayType, prevProps.widget.displayType) ||
       !isEqual(widget.interval, prevProps.widget.interval) ||
       !isEqual(widgetQueries, prevWidgetQueries) ||
-      !isSelectionEqual(selection, prevProps.selection)
+      !isSelectionEqual(selection, prevProps.selection) ||
+      !isEqual(cursor, prevProps.cursor)
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx:153]: do we need a deep comparison here?

Comment (edwardgou-sentry) [reply static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx]: probably not necessary, I've updated to just `!==`!

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**